### PR TITLE
add crawl_profile config to set custom CrawlProfile implementation

### LINKED ISSUE #90: Crawling site with subdomains
Hi,

For example, my site is `example.com`, and I have a subdomain `another.example.com`.
When I have this code: `SitemapGenerator::create($this->url)->writeToFile(...)`, the pages from `another.example.com` are not included in the sitemap. There are links to `another.example.com` on `example.com`.

Is there an option to include subdomains to the crawler maybe? 

Thanks! :)

### PR DESCRIPTION
**Pull request by fetzi**
fixes #90

### CODE REVIEW

**Comment by fetzi** [comment]
@freekmurze I've no idea what's breaking the PHP 7.2 build. Do you have an idea why the ExceptionHandler is not instantiable?

File: src/SitemapServiceProvider.php [hunk 1]
```diff
@@ -27,6 +28,10 @@ public function boot()
             ->give(function () {
                 return Crawler::create(config('sitemap.guzzle_options'));
             });
+
+        $this->app->bind(Profile::class, function ($app, $params) {
+            return new Profile(reset($params));
```
Comment (freekmurze) [inline src/SitemapServiceProvider.php]: Why do you use `reset` here?

Comment (fetzi) [reply src/SitemapServiceProvider.php]: I generally use it to return the first element of the array, but I can change to `$params[0]` if you want?

Comment (freekmurze) [reply src/SitemapServiceProvider.php]: Yeah, `params[0]` is more clear to me, I'll change this myself.

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**